### PR TITLE
[pkg/otlp] Document yaml option and environment variable for max OTLP gRPC message receive size

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

Documents `max_recv_msg_size_mib` and new environment variable (`DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_GRPC_MAX_RECV_MSG_SIZE_MIB`) to configure the maximum size of messages accepted by the OTLP gRPC endpoint.

### Describe how to test/QA your changes

Use [telemetrygen](https://github.com/liustanley/opentelemetry-collector-contrib/tree/stanley.liu/telemetrygen-update) to test the OTLP ingest with varying load sizes and without batching:
- ./telemetrygen traces --otlp-insecure --traces 1 --batch false --size 1 (should send)
- ./telemetrygen traces --otlp-insecure --traces 1 --batch false --size 3 (should send)
- ./telemetrygen traces --otlp-insecure --traces 1 --batch false --size 4 (should be dropped)

Then, try changing `max_recv_msg_size_mib` in both the datadog.yaml and the environment variable and verify tests work as intended.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by dineshg13 — APPROVED** [review]

**Review by estherk15 — APPROVED** [review]

**Comment by songy23** [comment]
Did QA with a custom build of Agent with this PR. Results are as expected in the QA instructions.

**Review by mx-psi — COMMENTED** [review]
I think this is the way to go IF we decide to do this, but I am sad to see us deviate from the upstream OTLP receiver and we need to have a very strong reason to do this. It should be very easy to move between the OpenTelemetry Collector and the Datadog Agent and in the past we have decided against making changes to the Agent to be as close as possible to the OTLP receiver.

My suggested alternative is to document `max_recv_msg_size_mib` in our config template, add an environment variable to be able to configure it, and ask users for which the 4MB default is not enough to modify it themselves. The 4MB default should be enough for most users, and once we bump it up it will be very hard to revert the change, so we need to think carefully about whether we should do this.

**Comment by liustanley** [comment]
After discussing with the team, we've decided against changing the default max receive size and instead will document this and add the environment variable to configure it.

File: cmd/trace-agent/config/config.go [hunk 1]
```diff
@@ -285,6 +285,7 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 		BindHost:               c.ReceiverHost,
 		GRPCPort:               grpcPort,
 		MaxRequestBytes:        c.MaxRequestBytes,
+		MaxGrpcRequestMib:      coreconfig.Datadog.GetFloat64("otlp_config.receiver.protocols.grpc.max_recv_msg_size_mib"),
```
Comment (mx-psi) [inline cmd/trace-agent/config/config.go:288]: I don't get why we need this. Otherwise it looks good to me :)

**Review by ogaca-dd — APPROVED** [review]

File: cmd/trace-agent/config/config.go (hunk 1 above)
Comment (liustanley) [reply cmd/trace-agent/config/config.go:288]: I tested more without this and it seems to work fine, so going to remove this

**Review by mx-psi — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]
👍 for Agent APM

**Pull request merged**